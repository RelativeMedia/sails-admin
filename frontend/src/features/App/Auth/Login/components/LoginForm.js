import React from 'react';
import PropTypes from 'prop-types';
import {Field, reduxForm} from 'redux-form';
import Form from 'components/Form';

import './styles.scss';
class LoginFormComponent extends React.Component {
  static propTypes = {
    invalid: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
    submitSucceeded: PropTypes.bool.isRequired,
    submitFailed: PropTypes.bool.isRequired,
    submit: PropTypes.func.isRequired,
    reset: PropTypes.func.isRequired,
  }
  render () {

    const {
      invalid,
      submitting,
      submit,
      reset
    } = this.props;

    return (<div className='LoginFormComponent'>
      <form onSubmit={submit}>
        <div className='col-lg-offset-3 col-lg-6'>
          <div className='panel panel-default'>
            <div className='panel-heading'>
              <h3 className='panel-title'>Sign In</h3>
            </div>
            <div className='panel-body'>
              <Field
                name='identity'
                label='Username or Email'
                placeholder='gordon.freeman@srpnet.com'
                inputLg
                component={Form.TextField}
                required
              />
              <Field
                name='password'
                label='Password'
                inputLg
                toggle
                required
                component={Form.PasswordField}
              />
              <div className='row'>
                <div className='col-lg-4'>
                  <Field
                    name='loginType'
                    label='Login Type'
                    component={Form.SelectField}
                    defaultValue='local'
                    options={[
                      {
                        value: 'local',
                        label: 'Local'
                      },
                      {
                        value: 'ad',
                        label: 'AD'
                      }
                    ]}
                  />
                </div>
              </div>
            </div>
            <div className='panel-footer'>
              <div className='btn-group'>
                <button
                className='btn btn-lg btn-default'
                type='button'
                disabled={submitting}
                onClick={reset}
                >
                  Cancel
                </button>
                <button
                  className='btn btn-lg btn-primary'
                  type='submit'
                  disabled={invalid || submitting}
                  onClick={submit}
                >
                  {(submitting)
                    ? <i className='fa fa-spin fa-spinner' />
                    : 'Sign In'
                  }
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>);
  }
}

const validate = (values) => {
  let errors = {};
  if (!values.identity) {
    errors.identity = 'Identity is Required';
  }

  if (!values.password) {
    errors.password = 'Password is Required';
  }

  return errors;
};

export default reduxForm({
  form: 'LoginForm',
  validate
})(LoginFormComponent);
