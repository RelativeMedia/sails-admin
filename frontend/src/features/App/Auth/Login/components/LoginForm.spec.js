import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('<LoginForm />', () => {
  it('should render the login form', () => {
    const wrapper = shallow(<LoginForm onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
