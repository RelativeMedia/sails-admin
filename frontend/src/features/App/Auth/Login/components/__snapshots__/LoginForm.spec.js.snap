// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render the login form 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    onSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "LoginForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "onSubmit": [Function],
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "validate": [Function],
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "LoginForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "onSubmit": [Function],
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "validate": [Function],
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
